odoo/addons/base/models/ir_module.py

```diff
@@ -418,7 +419,11 @@ def must_install(module):
             modules._state_update('to install', ['uninstalled'])
 
             # Determine which auto-installable modules must be installed.
-            modules = self.search(auto_domain).filtered(must_install)
+
+            if 'skipautoinstall' in (config.get('dev_mode') or ''):
```

Comment:
```suggestion
            if 'skipautoinstall' in config['dev_mode']:
```
I looks like we always set this key to a list
https://github.com/odoo/odoo/blob/b093786714e9e8567cf75abf78ac3d954a3d89b2/odoo/tools/config.py#L556

Reply:
> If you think that it is not a big deal to break this if it exists somewhere, I can apply your suggestion.

I agree with you. Better be cautious. You can disregard my (nit) suggestions. :)